### ISSUE TITLE
form controls: focus lost upon toggling disabled mode

### ISSUE THREAD

**Issue by florian-sanders-cc** [CleverCloud/clever-components#1102]
## Context

During async API calls, we disable form controls.
Since disabled form controls cannot receive focus, the focus is lost.

When users are submitting the form using the submit button, this does not create any issue because the <cc-button> element does not actually disable the inner <button> element (we use `aria-disabled` and prevent clicks from doing anything).

When users are submitting the form by pressing enter within a form control, this creates an issue because focus goes nowhere, it is lost.

We may need to consider the same solution as with the `cc-button` component and go for `aria-disabled` + make the form control unresponsive but this is trickier than with the `cc-button` because there are more interactions to disable.

## Decision

We have decided to remain fairly close to the native HTML behavior:

- Use `disabled` only when we truly want to disable a form control,
- Use `readonly` during API calls because it preserves focus.

## TODO

- [ ] update the docs of `readonly` and `disabled` props to include some guidance about when to use which,
- [ ] change all occurrences of `disabled` during an API call to `readonly`.

**Comment by florian-sanders-cc** [comment]
Note: I forgot to specify that this issue depends on the browser used to reproduce because not all of them makes disabled form control unfocusable :upside_down_face:
